templates/rsyslog.conf.erb

```diff
@@ -87,7 +87,9 @@ $ActionQueueSaveOnShutdown on # save messages to disk on shutdown
 $ActionQueueType LinkedList   # run asynchronously
 $ActionResumeRetryCount -1    # infinite retries if host is down
 <% end -%>
-<%= @source_facilities %> <% if @transport_protocol == 'tcp' -%>@<% end -%>@<%= @log_server -%>:<%= @log_server_port %>
+<% for server in @log_server -%>
```

Comment:
I would recommend to move the logic into Puppet code. See my PR https://github.com/sharlander/puppet-module-rsyslog/pull/1 which ensures that $log_server is an array.